roxie/ccd/ccdfile.cpp

```diff
@@ -1128,11 +1128,20 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
         {
             CriticalBlock b(crit);
             Linked<ILazyFileIO> f = files.getValue(localLocation);
-            if (f && f->isAlive())
+            while (f && f->isAlive())
```

Comment:
I'm not sure I like this - you never go around this more than once, it is only here so you can use continue to break out of the loop.   Better to add an if (!f) test just after the test of ROXIE_KEY